docs/conf.py

```diff
@@ -296,3 +302,85 @@ def setup(app):
 if read_the_docs_build:
 
     subprocess.call('cd ../minorminer/docs/; make cpp', shell=True)
+
+github_map = {'dwavebinarycsp': 'dwavebinarycsp',
+              'cloud': 'dwave-cloud-client',
+              'dimod':  'dimod',
+              'dwave_networkx': 'dwave-networkx',
+              'greedy': 'dwave-greedy',
+              'hybrid': 'dwave-hybrid',
+              'inspector': 'dwave-inspector',
+              'minorminer': 'minorminer',
+              'neal': 'dwave-neal',
+              'penaltymodel': {'cache': 'penaltymodel_cache',
+                               'core': 'penaltymodel_core',
+                               'lp': 'penaltymodel_lp',
+                               'maxgap': 'penaltymodel_maxgap',
+                               'mip': 'penaltymodel_mip'},
+              'system': 'dwave-system',
+              'embedding': 'dwave-system',
+              'tabu': 'dwave-tabu'}
+
+def linkcode_resolve(domain, info):
+    """
+    Find the URL of the GitHub source for dwave-ocean-sdk objects.
+    """
+    # Based on https://github.com/numpy/numpy/blob/main/doc/source/conf.py
+    # Updated to work on multiple submodules and fall back to next-level 
+    # module for objects such as properties
+
+    if domain != 'py':
+        return None
+
+    obj={}
+    obj_inx = 0
+    obj[obj_inx] = sys.modules.get(info['module'])
+    for part in info['fullname'].split('.'):
+        obj_inx += 1
+        try:
+            obj[obj_inx] = getattr(obj[obj_inx - 1], part)
+        except Exception:
+            pass
+
+    # strip decorators, which would resolve to the source of the decorator
+    # https://bugs.python.org/issue34305
+    for i in range(len(obj)):
+           obj[i] = inspect.unwrap(obj[i])
+
+    for i in range(len(obj)-1, -1, -1): 
+        try: 
+           fn = inspect.getsourcefile(obj[i]) 
+           if fn: 
+              obj_inx = i
+              break 
+        except:
+           pass 
+
+    linespec = ""
+    try:
+        source, lineno = inspect.getsourcelines(obj[obj_inx])
+        linespec = "#L%d" % (lineno)
+    except Exception:
+        linespec = ""
+
+    if not fn or not "site-packages" in fn:
+       return None
+    
+    if ".egg" in fn:
+       fn = fn.replace(fn[:fn.index("egg")+len("egg")], "")   
+    else:
+       fn = fn.replace(fn[:fn.index("site-packages")+len("site-packages")], "") 
+
+    target = fn.split("/")
+    repo = target[1] if not target[1] == "dwave" else target[2]
+
+    if repo in ["minorminer", ]:
+        fn = "https://github.com/dwavesystems/" + github_map[repo] + "/blob/main" + fn
+    elif repo == 'penaltymodel':
+        fn = "https://github.com/dwavesystems/penaltymodel/blob/master/" +  \
+             github_map[repo][info['module'].split('.')[1]] + fn  
+    else:
+        fn = "https://github.com/dwavesystems/" + github_map[repo] + "/blob/master" + fn     
```

Comment:
You can just [use `HEAD` to point to tip of the default branch](https://stackoverflow.com/questions/64726262/is-there-a-stable-url-to-always-get-the-default-branch-in-github).

Reply:
However, shouldn't we use the _actual version_ of the file included in current Ocean, and not the latest master?

Reply:
@randomir, I'll still clean up the updated code but looks like your proposal to use `python  setup.py egg_info` rather than `distutils.core.run_setup("setup.py")` will work.

Reply:
Okay, now links to the version of source specified by the SDK.